docs/testing.md

```diff
@@ -0,0 +1,114 @@
+# Testing Taps & Targets
+
+The Meltano SDK includes suites of standard tests for both Taps and Targets to help you get started.
+These suites cover most common cases out-of-the-box, and tests are added to the standard suites as new errors are encountered by users in their deployments.
+
+## Test Framework
+
+The Meltano SDK test framework consists of 4 main components:
+
+1. A runner class (`TapTestRunner` and `TargetTestRunner`), responsible for executing Taps/Targets and capturing their output.
+1. A suite dataclass, containing a list of tests.
+1. A test template classes (`TapTestTemplate`, `StreamTestTemplate`, `AttributeTestTemplate` and `TargetTestTemplate`), with methods to `.setup()`, `.test()`, `.validate()` and `.teardown()` (called in that order using `.run()`).
+1. A `get_test_class` factory method, to take a runner and a list of suits and return a `pytest` test class.
+
+## Example Usage
+
+If you created your Tap/Target using the provided cookiecutter templates, you will find the following snippets in `<library_name>/tests/test_core.py`.
+
+### Testing Taps
+
+```python
+import datetime
+
+from singer_sdk.testing import (
+    TapTestRunner,
+    get_test_class
+)
+from singer_sdk.testing.suites import (
+    tap_stream_attribute_tests,
+    tap_stream_tests,
+    tap_tests,
+)
+
+from example.tap import TapExample
+
+SAMPLE_CONFIG = {
+    "start_date": datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%d")
+}
+
+
+# Run standard built-in tap tests from the SDK:
+TestTapExample = get_test_class(
+    test_runner=TapTestRunner(tap_class=TapExample, config=SAMPLE_CONFIG),
+    test_suites=[tap_tests, tap_stream_tests, tap_stream_attribute_tests],
+)
+```
```

Comment:
It's a bit of boilerplate and imports just for generating the standard tests for a tap, so wdyt about making the interface a bit simpler by wrapping `get_test_class` into `get_tap_test_class` and `get_target_test_class`:

```python
# in singer_sdk.testing.factory
def get_tap_test_class(
    tap_class,
    config=None,
    *,
    tap_tests=True,
    stream=True,
    stream_attributes=True,
):
    # or use enum flags
    suites = []
    if tap_tests:
        suites.append(tap_tests)
    if stream:
        suites.append(tap_stream_tests)
    if stream_attributes:
        suites.append(tap_stream_attribute_tests)

	return get_test_class(
        test_runner=TapTestRunner(tap_class=tap_class, config=config),
        test_suites=suites,
    )
        
# in the examples and cookiecutter
TestTapExample = get_tap_test_class(
    tap_class=TapExample,
    config=SAMPLE_CONFIG,
    tap_tests=True,         # adds `tap_tests` under the hood
    stream=True,            # adds `tap_stream_tests` under the hood
    stream_attributes=True, # adds `tap_stream_attribute_tests` under the hood
)
```

Reply:
Great idea, I have added `get_tap_test_class` and `get_target_test_class` as suggested 🙏✅